cli/util/util.go

```diff
@@ -26,3 +31,39 @@ func Dial(addr, certPEM string) (*grpc.ClientConn, error) {
 		grpc.WithDefaultCallOptions(grpc.UseCompressor(gzip.Name)),
 		grpc.WithUnaryInterceptor(errors.UnaryClientInterceptor))
 }
+
+func GetDockerClient() (*client.Client, error) {
+	dockerClient, dcErr := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+
+	// If a custom host is set, don't try any funny business.
+	if dcErr != nil || dockerClient.DaemonHost() != client.DefaultDockerHost {
+		return dockerClient, dcErr
+	}
+
+	socketUrl, err := url.Parse(dockerClient.DaemonHost())
+	if err != nil {
+		// We really do not expect this to ever happen.
+		panic(err)
+	}
+
+	if socketUrl.Scheme != "unix" {
+		return dockerClient, dcErr
+	}
+
+	// Check to see if this unix socket actually exists.
+	f, err := os.Open(socketUrl.Path)
```

Comment:
When would creating the client succeed if the socket doesn't exist? I think I just don't understand the reason for this patch / the Docker client creation API.

Reply:
Creating the client does not open the socket. The error does not happen until we try to build, which is pretty annoying.

Reply:
I see. Does the client just check that the path exists or something?